### DISCUSSION TITLE
ESP32 modules compatable with sensESP

### DISCUSSION THREAD

**Discussion by gregsyoung** [SignalK/SensESP#788] [General]
there is a growing proliferation of  "ESP32"modules - as newer espressif chips are released et al.

I recently went to purchase some more modules, and was bewildered .. and of course naming convention & clarity from online seller adverts .. further complicates matters.

I looked in the sensESP Github documentation.... it says "This can be done with commonly available ESP32DevKit boards " 
.... however not sure if thats sufficiently descriptive ?
likewise in sensESP documentation it says "However, any ESP32 board will work fine" - is this correct?
eg will it work on a ESP32-C3 board?

originally i thought the module needed to have a WROOM32 chipset? (one of its numerous variants )?
.. but now im confused :) 

maybe there needs to be additional spec that needs to be added to the "what hardware will sensESP support" type question?

**Comment by mairas** [comment]
That's a valid point a a reasonable thing to be confused about. 😁 I guess that text was written when there was just the basic ESP32-WROOM32 module available.

I am doing most of my development on SH-ESP32 and HALMET boards which use the ESP32-WROOM32 modules. Boards using the original ESP32 or ESP32-S2/S3 should be expected to have the least issues. However, I do have a strong interest in supporting the RISC-V based modules (ESP32-Cx, SP32-P4) as well. There might be minor issues with HW interfaces such as ADC and PWM, but probably nothing that couldn't be easily fixed.
